.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,60 @@
+name: To Do app CI/CD
+on: [push]
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    env:
+        TOKEN: ${{ secrets.TOKEN }}
+        SECRET_KEY: ${{ secrets.SECRET_KEY }}
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Build
+      run: docker build --target test --tag todo-app:test .
+
+    - name: Unit/Integration Test
+      run: docker run todo-app:test tests
+    
+    - name: End to End Test
+      run: docker run -e SECRET_KEY -e TOKEN todo-app:test tests_e2e
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    env:
+        docker_hub_password: ${{ secrets.docker_hub_password }}
```

Comment:
I know Hugh gave you this, but I'd suggest keeping these to the standard UPPERCASE casing for environment variables!